Milliner/src/Service/MillinerService.php

```diff
@@ -554,39 +527,86 @@ public function saveExternal(
+                $fedora_url,
+                $timestamp,
+                null,
+                $headers
+            );
+            $status = $response->getStatusCode();
+            if (!in_array($status, [201])) {
+                $reason = $response->getReasonPhrase();
+                throw new \RuntimeException(
+                    "Client error: `POST $fedora_url` resulted in `$status $reason` response: " .
+                    $response->getBody(),
+                    $status
                 );
-                $status = $response->getStatusCode();
-                if (!in_array($status, [201])) {
-                    $reason = $response->getReasonPhrase();
-                    throw new \RuntimeException(
-                        "Client error: `POST $fedora_url` resulted in `$status $reason` response: " .
-                        $response->getBody(),
-                        $status
-                    );
-                }
-                // Return the response from Fedora.
-                return $response;
-            } catch (Exception $e) {
-                $this->log->error('Caught exception when creating version: ', $e->getMessage(), "\n");
             }
-        } else {
-            return new Response(404);
+            // Return the response from Fedora.
+            return $response;
+        } catch (Exception $e) {
+            $this->log->error('Caught exception when creating version: ', $e->getMessage(), "\n");
         }
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getFileFromMedia(
+        $source_field,
+        $json_url,
+        $token = null
+    ) {
+        // First get the media json from Drupal.
+        $headers = empty($token) ? [] : ['Authorization' => $token];
+        $drupal_response = $this->drupal->get(
+            $json_url,
+            ['headers' => $headers]
+        );
+
+        $jsonld_url = $this->getLinkHeader($drupal_response, "alternate", "application/ld+json");
+        $media_json = json_decode(
+            $drupal_response->getBody(),
+            true
+        );
+
+        if (!isset($media_json[$source_field]) || empty($media_json[$source_field])) {
+            throw new \RuntimeException(
+                "Cannot parse file UUID from $json_url.  Ensure $source_field exists on the media and is populated.",
+                500
+            );
+        }
+        $file_uuid = $media_json[$source_field][0]['target_uuid'];
+
+        // Get the file's LDP-NR counterpart in Fedora.
+        $urls = $this->gemini->getUrls($file_uuid, $token);
+        if (empty($urls)) {
+            $file_url = $media_json[$source_field][0]['url'];
+            throw new \RuntimeException(
+                "$file_url has not been mapped in Gemini with uuid $file_uuid",
+                404
+            );
+        }
+        return array('fedora'=>$urls['fedora'], 'jsonld' =>$jsonld_url, 'drupal'=>$urls['drupal']);
```

Comment:
I hate myself for nitpicking on this, but you should use `[]` instead of `array` here